packages/http/package.json

```diff
@@ -23,6 +23,8 @@
     "ajv-oai": "^1.0.0",
     "axios": "^0.18.0",
     "caseless": "^0.12.0",
+    "fp-ts": "^1.18.2",
```

Comment:
I hate to say this but `fp-ts` seem to require some work on the documentation end. Cool to use new tools but just FYI it may be a steep learning curve for contributors.

Reply:
Personally, when I saw it was a big positive! I didn't use `fp-ts`, but I'd like to see fp used more in our codebase (even if it was only limiting side effects). I'm all for keepking `fp-ts`. This is clearly a good sign.

Reply:
> some work on the documentation end

This is true in general, the documentation is lacking. Speaking with the authors they're working on it as part of the next release but, in general, it's not the library documentation you need, but a mini training on the concepts. I'll have a presentation about this in [August in Columbus](https://www.javascriptandfriends.com/speakers), you might want to watch the recording :trollface:

> but I'd like to see fp used more in our codebase (even if it was only limiting side effects)

It's great to have you into this stuff! My ultimate goal would be to make Prism a functional mock server if possible. For now I started with logging which is a small project and the risk is contained. As you probably noticed, at a certain point we need to go out from the functional world and throw an exception, but as this "virus" expands, little by little the `throw` statements will hopefully start to disappear!

Reply:
@XVincentX, so very cool. From `Professor Frisby's Mostly Adequate Guide To Functional Programming`:
```
It may come as a shock, but throw/catch is not very pure. When an error is thrown, instead of returning an output value, 
we sound the alarms! The function attacks, spewing thousands of 0's and l's like shields & spears in an electric battle 
against our intruding input. With our new friend Either , we can do better than to declare war on input, we can respond with 
a polite message.
```
And I do know there is `Either` somewhere in this PR! ;)